raft/rawnode.go

```diff
@@ -170,7 +187,10 @@ func (rn *RawNode) HasReady() bool {
 	if r.raftLog.hasNextUnstableSnapshot() {
```

Comment:
Should we order this one down? After all, this is a rare case, so we don't have to run it in the common path. Checking `r.msgs`, `r.msgsAfterAppend` should probably be first, followed by hard state.

Also, `r.softState()` probably allocates on each `HasReady` call, which seems dumb. We should make that method return a value and also order it way down since it's basically not usually true.

Not really what you're after in this PR so maybe entomb this comment as a TODO instead.

Reply:
Done.